keps/sig-node/4742-node-topology-downward-api/README.md

```diff
@@ -0,0 +1,411 @@
+
+### Version Skew Strategy
+
+There should be no risks with version skew between control plane and nodes since there's no changes made to the kubelet's downward API implementation.
+All changes proposed in this KEP are control-plane only.
+
+For n-3 kubelets, old kubelets still support consuming Pod labels via downward API. This capability should be supported as long as the control plane
+has this feature enabled.
+
+There may be some risks for version skew with other components (e.g CSI or CNI) if the downward API is used
+to pass topology information to those components. Components should ensure they are using a kube-apiserver version
+that supports copying topology labels to Pods.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+A feature gate `PodTopologyLabelsAdmission` will be added to the kube-apiserver to enable or disable this feature.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `PodTopologyLabelsAdmission`
+  - Components depending on the feature gate: `kube-apiserver`
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+Yes, all Pods will have topology labels copied over.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes it can, but Pods created while the feature is enabled will continue to have topology labels.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+New Pods will have topology labels copied over. Older Pods created while the feature was disabled will be missing topology labels.
+
+###### Are there any tests for feature enablement/disablement?
+
+Tests will be added to ensure feature gate works as expected.
+
+### Rollout, Upgrade and Rollback Planning
+
+TODO for Beta.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Failure on rollout is unlikely since we do not override any existing topolgoy labels on Pods.
+Already running workloads that is already copying topology labels to Pods should be untouched by the new admission plugin.
+
+###### What specific metrics should inform a rollback?
+
+N/A
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Upgrade / rollback will be manually tested before graduation to Beta.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+TODO for GA.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Look for topology labels in Pod labels.
```

Comment:
There is actually no user control over the feature so this question is N/A.

Reply:
done